src/Guide/Api/Methods.hs

```diff
@@ -181,27 +230,25 @@ createTrait db itemId traitType text = do
 -- | Update the text of a trait (pro/con).
 setTrait :: DB -> Uid Item -> Uid Trait -> CTextEdit -> Handler NoContent
 setTrait db itemId traitId CTextEdit{..} = do
-  dbQuery db (GetItemMaybe itemId) >>= \case
-    Nothing -> throwError (err404 {errBody = "Item not found"})
-    Just _ -> do
-      dbQuery db (GetTraitMaybe itemId traitId) >>= \case
-        Nothing -> throwError (err404 {errBody = "Trait not found"})
-        Just Trait{..} -> do
-          let serverModified = markdownInlineMdSource _traitContent
-          let original = unH cteOriginal
-          let modified = unH cteModified
-          if original /= serverModified then do
-            let merged = merge original modified serverModified
-            let conflict = CMergeConflict
-                    { cmcOriginal = cteOriginal
-                    , cmcModified = cteModified
-                    , cmcServerModified = H serverModified
-                    , cmcMerged = H merged
-                    }
-            throwError (err409 {errBody = encode conflict})
-          else do
-            (_edit, _newCategory) <- dbUpdate db (SetTraitContent itemId traitId modified)
-            pure NoContent
+  _ <- getItemOrFail db itemId
+  dbQuery db (GetTraitMaybe itemId traitId) >>= \case
+    Nothing -> throwError $ err404 {errBody = "Trait not found"}
```

Comment:
It would be nice to have "getTraitOrFail" that would check for both item's existence and trait's existence